libs/features/jarvis-chat.js

```diff
@@ -0,0 +1,343 @@
+    open({
+      sourceType,
+      sourceText,
+    });
+  } else {
+    open({});
+  }
+};
+
+const startInitialization = async (config, event) => {
+  const asset = 'https://client.messaging.adobe.com/latest/AdobeMessagingClient';
+  await Promise.all([
+    loadStyle(`${asset}.css`),
+    loadScript(`${asset}.js`),
+  ]);
+  let language;
+  let region;
+  if (config.locale.ietf.includes('-')) {
+    [language, region] = config.locale.ietf.split('-');
+  } else {
+    [region, language] = config.locale.prefix.replace('/', '').split('_');
+  }
+
+  window.AdobeMessagingExperienceClient.initialize({
+    appid: config.jarvis.id,
+    appver: config.jarvis.version,
+    env: config.env.name !== 'prod' ? 'stage' : 'prod',
+    clientId: window.adobeid?.client_id,
+    accessToken: window.adobeIMS?.isSignedInUser()
+      ? `Bearer ${window.adobeIMS.getAccessToken()?.token}` : undefined,
+    lazyLoad: true,
+    loadedVia: 'milo',
+    language: language || 'en',
+    region,
+    cookiesEnabled: window.adobePrivacy?.activeCookieGroups()?.length > 1,
+    cookies: {
+      mcid: window.alloy ? await window.alloy('getIdentity')
+        .then((data) => data?.identity?.ECID) : undefined,
+    },
+    callbacks: {
+      initCallback: (data) => {
+        chatInitialized = !!data?.releaseControl?.showAdobeMessaging;
+      },
+      onReadyCallback: () => {
+        if (config.jarvis.onDemand) {
+          openChat(event);
+        }
+      },
+      initErrorCallback: () => {},
+      chatStateCallback: () => {},
+      getContextCallback: () => {},
+      signInProvider: window.adobeIMS?.signIn,
+      analyticsCallback: (eventData) => {
+        if (!window.alloy_all || !window.digitalData) return;
+        const data = eventData?.events?.[0]?.data;
+        if (!data || isSilentEvent(data)) return;
+        if (data['event.subcategory']?.toLowerCase() === 'launch'
+          && data['event.workflow']?.toLowerCase() === 'init') {
+          if (data['event.type']?.toLowerCase() === 'render') {
+            sendChatIconRenderEvent(data);
+            return;
+          }
+          if (data['event.type']?.toLowerCase() === 'click') {
+            sendChatIconClickEvent(data);
+            return;
+          }
+        }
+        if (data['content.name']?.toLowerCase() === 'auth-subproduct') {
+          sendProductEvent(data);
+          return;
+        }
+        if (data['content.name']?.toLowerCase() === '5-star-survey') {
+          sendSurveyFeedbackEvent(data);
+          return;
+        }
+        if (data['event.error_code'] && data['event.error_type']) {
+          sendChatErrorEvent(data);
+          return;
+        }
+        sendPrimaryEvent(data);
+      },
+    },
+  });
+};
+
+const initJarvisChat = async (config, loadScriptFunction, loadStyleFunction) => {
+  if (!config?.jarvis) return;
+
+  loadScript = loadScriptFunction;
+  loadStyle = loadStyleFunction;
+
+  document.addEventListener('click', async (event) => {
```

Comment:
Does jarvis also load late?  Really wonder if there's a better soln but don't know what that is